paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/PaymentOptionFactory.kt

```diff
@@ -44,10 +46,19 @@ internal class PaymentOptionFactory @Inject constructor(
         // Some payment options don't have an icon URL, and are loaded locally via resource.
         val lightThemeIconUrl = paymentOption.lightThemeIconUrl
         val darkThemeIconUrl = paymentOption.darkThemeIconUrl
+
+        @Suppress("DEPRECATION")
+        val drawableIsMissing = paymentOption.drawableResourceId == MISSING_DRAWABLE_RESOURCE
+
         return if (isDarkTheme() && darkThemeIconUrl != null) {
             loadIcon(darkThemeIconUrl)
         } else if (lightThemeIconUrl != null) {
             loadIcon(lightThemeIconUrl)
+        } else if (drawableIsMissing) {
+            errorReporter.report(
+                ErrorReporter.UnexpectedErrorEvent.PAYMENT_OPTION_MISSING_ICON_URL_AND_RES,
+            )
+            throw IllegalStateException("Missing icon resource and icon URLs - can't load")
```

Comment:
I don't think we want to do this. We won't ever get the error reports. 

We probably just want to silently load a blank resource based image, which to the user would look like nothing.